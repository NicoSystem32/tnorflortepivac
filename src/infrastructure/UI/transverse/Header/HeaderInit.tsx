import Container from 'react-bootstrap/Container'
import Navbar from 'react-bootstrap/Navbar'

// Images
import { Logo } from '../../utils/getImages'

// Icons
import { HeadsetSVG, IconUsercogSVG, IconUnlockSVG } from '../../utils/getIcons'

import './header.scss'

const HeaderInit = (): JSX.Element => {
    return (
        <Navbar className="theme-header">
            <Container className="content-header">
                <a id="logo" className="logo" href="/login">
                    <img alt="Logo Cavipetrol" src={Logo} />
                </a>
                <a id="support-hd" className="options-content border-pipe" href="/support">
                    <img src={HeadsetSVG} alt="Servicio al asociado" className="options-img" />
                    <p className="options-text">Servicio al asociado</p>
                </a>
                <a
                    id="forget-password-hd"
                    className="options-content border-pipe"
                    href="/forget-password"
                >
                    <img src={IconUsercogSVG} alt="" className="options-img" />
                    <p className="options-text">Olvidé mi contraseña</p>
                </a>
                <a
                    id="unlocked-user-hd"
                    className="options-content"
                    href="https://transacciones.cavipetrol.com/DwPortalApp/Cavionline/UI/Web/bloqueado_m.aspx?tabID=417&mID=3411"
                    target="_blank"
                    rel="noreferrer"
                >
                    <img
                        alt=""
                        src={IconUnlockSVG}
                        srcSet={IconUnlockSVG}
                        className="options-img"
                    />

                    <p className="options-text">Desbloquear mi usuario</p>
                </a>
                <a
                    id="new-user-hd"
                    className="option-register"
                    href="https://transacciones.cavipetrol.com/DwPortalApp/Cavionline/UI/Web/x_crea0_m.aspx?tabID=417&mID=3411"
                    target="_blank"
                    rel="noreferrer"
                >
                    <span className="register"></span>
                </a>
            </Container>
        </Navbar>
    )
}

export default HeaderInit
