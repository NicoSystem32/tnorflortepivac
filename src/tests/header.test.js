import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import HeaderInit from '../infrastructure/UI/transverse/Header/HeaderInit'

describe('Pruebas en <HeaderInit />', () => {
    const wrapper = shallow(<HeaderInit />)

    test('debe de mostrar <HeaderInit /> correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('el logo debe redirigir correctamente', () => {
        expect(wrapper.find('#logo').prop('href')).toBe('/login')
    })

    test('boton de soporte debe redirigir correctamente', () => {
        expect(wrapper.find('#support-hd').prop('href')).toBe('/support')
    })

    test('boton de olvido de contraseÃ±a debe redirigir correctamente', () => {
        expect(wrapper.find('#forget-password-hd').prop('href')).toBe('/forget-password')
    })

    test('boton chatear debe redirigir correctamente', () => {
        expect(wrapper.find('#unlocked-user-hd').prop('href')).toBe(
            'https://transacciones.cavipetrol.com/DwPortalApp/Cavionline/UI/Web/bloqueado_m.aspx?tabID=417&mID=3411'
        )
    })

    test('boton chatear debe redirigir correctamente', () => {
        expect(wrapper.find('#new-user-hd').prop('href')).toBe(
            'https://transacciones.cavipetrol.com/DwPortalApp/Cavionline/UI/Web/x_crea0_m.aspx?tabID=417&mID=3411'
        )
    })
})
