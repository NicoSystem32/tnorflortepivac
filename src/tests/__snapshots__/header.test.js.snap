// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeaderInit /> debe de mostrar <HeaderInit /> correctamente 1`] = `
<Fragment>
  <Navbar
    className="theme-header"
    collapseOnSelect={false}
    expand={true}
    variant="light"
  >
    <Container
      className="content-header"
      fluid={false}
    >
      <a
        className="logo"
        href="/login"
        id="logo"
      >
        <img
          alt=""
          src="logo.png"
        />
      </a>
      <a
        className="options-content border-pipe"
        href="/support"
        id="support-hd"
      >
        <img
          alt="Servicio al asociado"
          className="options-img"
          src="headset.svg"
          srcSet="headset.png"
        />
        <p
          className="options-text"
        >
          Servicio al asociado
        </p>
      </a>
      <a
        className="options-content border-pipe"
        href="/forget-password"
        id="forget-password-hd"
      >
        <img
          className="options-img"
          src="users-cog.svg"
          srcSet="users-cog.png"
        />
        <p
          className="options-text"
        >
          Olvid√© mi usuario
        </p>
      </a>
      <a
        className="options-content"
        href="https://transacciones.cavipetrol.com/DwPortalApp/Cavionline/UI/Web/bloqueado_m.aspx?tabID=417&mID=3411"
        id="unlocked-user-hd"
        target="_blank"
      >
        <img
          className="options-img"
          src="unlock.svg"
          srcSet="unlock.png"
        />
        <p
          className="options-text"
        >
          Desbloquear mi usuario
        </p>
      </a>
      <a
        className="option-register"
        href="https://transacciones.cavipetrol.com/DwPortalApp/Cavionline/UI/Web/x_crea0_m.aspx?tabID=417&mID=3411"
        id="new-user-hd"
        target="_blank"
      >
        <span
          className="register"
        />
      </a>
    </Container>
  </Navbar>
</Fragment>
`;
